content/posts/2022-09-14-introducing-jaiminho/index.md

```diff
@@ -0,0 +1,136 @@
+---
+title: Introducing Jaiminho, an implementation of the Transactional Outbox Pattern for Django
+author: Rafael Zanetti
+twitter: LoadsmartEng
+layout: post
+lang: en
+path: /blog/introducing-jaiminho
+date: 2022-09-14
+comments: true
+---
+
+At Loadsmart, many of our services are organized within an event-driven architecture and face some usual challenges that come with such an architecture. One of them is **dual writes**, which happens when an application has to change data on two different places. A common example of dual writes is an application that needs to store data on a database and notify that change by emitting an event which may be consumed by different services. The inherent issue with this scenario is that if one of these two operations fails, the producer and consumer applications will have data inconsistencies, because the consumer never received the event informing the database change.
```

Comment:
`the producer and consumer applications will have data inconsistencies, because the consumer never received the event informing the database change.`

Would be nice to describe the problem when we successfully publish the event first, and them, fail to commit on local DB. When people don't know the dual writes issue, they use to propose this solution of publish the event first

Reply:
Added the description! Can you check if it was what you were thinking?